pkg/api/controller.go

```diff
@@ -373,6 +364,117 @@ func (c *Controller) StartBackgroundTasks(reloadCtx context.Context) {
 	}
 
 	if c.Config.Extensions != nil {
-		ext.EnableScrubExtension(c.Config, c.StoreController, c.Log)
+		ext.EnableScrubExtension(c.Config, c.Log, false, nil, "")
+	}
+
+	c.StartPeriodicTasks()
+}
+
+func (c *Controller) StartPeriodicTasks() {
+	// start periodic gc and/or scrub for DefaultStore
+	go StartPeriodicTasksForImageStore(c.StoreController.DefaultStore, c.Config.Storage.GC, c.Config.Storage.GCInterval,
+		c.Config.Extensions, c.Log)
+
+	if c.Config.Storage.SubPaths != nil {
+		for route, storageConfig := range c.Config.Storage.SubPaths {
+			// Enable running garbage-collect or/and scrub periodically for subImageStore
+			go StartPeriodicTasksForImageStore(c.StoreController.SubStore[route], storageConfig.GC, storageConfig.GCInterval,
+				c.Config.Extensions, c.Log)
+		}
+	}
+}
+
+func StartPeriodicTasksForImageStore(imageStore storage.ImageStore, configGC bool, configGCInterval time.Duration,
+	extensions *extconf.ExtensionConfig, log log.Logger,
+) {
+	var interval time.Duration
+
+	scrubInterval := time.Duration(0)
+	gcInterval := time.Duration(0)
+
+	gc := false
+	scrub := false
+
+	if configGC && configGCInterval != 0 {
+		gcInterval = configGCInterval
+		gc = true
+	}
+
+	if extensions != nil && extensions.Scrub != nil && extensions.Scrub.Interval != 0 {
+		scrubInterval = extensions.Scrub.Interval
+		scrub = true
+	}
+
+	switch {
+	case scrub && !gc:
+		interval = scrubInterval
+	case !scrub && gc:
+		interval = gcInterval
+	case scrub && gc && scrubInterval <= gcInterval:
+		interval = scrubInterval
+	case scrub && gc && gcInterval < scrubInterval:
+		interval = gcInterval
+	default:
+		return
+	}
+
+	var lastGC, lastScrub time.Time
+
+	for {
+		log.Info().Msg(fmt.Sprintf("Starting periodic background tasks(gc and/or scrub) for %s", imageStore.RootDir()))
+
+		// Enable running garbage-collect or/and scrub periodically for imageStore
+		RunGCScrubPeriodically(imageStore, gc, scrub, log)
+
+		log.Info().Msg(fmt.Sprintf("Finishing periodic background tasks(gc and/or scrub) for %s", imageStore.RootDir()))
+
+		if gc {
+			lastGC = time.Now()
+		}
+
+		if scrub {
+			lastScrub = time.Now()
+		}
+
+		time.Sleep(interval)
+
+		if !lastGC.IsZero() && time.Since(lastGC) >= gcInterval {
+			gc = true
+		}
+
+		if !lastScrub.IsZero() && time.Since(lastScrub) >= scrubInterval {
+			scrub = true
+		}
+	}
+}
+
+func RunGCScrubPeriodically(imgStore storage.ImageStore, gc, scrub bool, log log.Logger) {
+	repos, err := imgStore.GetRepositories()
+	if err != nil {
+		log.Error().Err(err).Msg(fmt.Sprintf("error while running background task for %s", imgStore.RootDir()))
+	}
+
+	for _, repo := range repos {
+		if gc {
```

Comment:
I already check it in for loop from StartPeriodicTasksForImageStore after each interval
`if !lastGC.IsZero() && time.Since(lastGC) >= gcInterval {
	gc = true
}`